org.wso2.carbon.metadata.server/src/main/java/org/wso2/carbon/metadata/server/dao/inmemory/InMemoryMetadataDAO.java

```diff
@@ -16,10 +16,139 @@
 
 package org.wso2.carbon.metadata.server.dao.inmemory;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.metadata.server.api.Key;
+import org.wso2.carbon.metadata.server.api.MetadataStoreException;
+import org.wso2.carbon.metadata.server.api.Resource;
 import org.wso2.carbon.metadata.server.dao.MetadataDAO;
+import org.wso2.carbon.metadata.server.impl.CollectionImpl;
+import org.wso2.carbon.metadata.server.impl.ResourceImpl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import javax.xml.ws.http.HTTPException;
 
 /**
  * In Memory Metadata DAO
  */
 public class InMemoryMetadataDAO implements MetadataDAO {
+
+    private static final Logger logger = LoggerFactory.getLogger(InMemoryMetadataDAO.class);
+    //stores resources against the path
+    private static HashMap<String, Resource> inMemoryStore;
```

Comment:
Any reason for not using a generic type here i.e Map<?,?> instead of HashMap<?,?> ?